### PR TITLE
Fixes #35701 - Export button should use selectable columns

### PR DESCRIPTION
**Pull request by pkoprda**
Export button on the host index page should export columns selected by column selector.

### CODE REVIEW

File: app/controllers/hosts_controller.rb [hunk 1]
```diff
@@ -912,7 +912,7 @@ def host_attributes_for_templates(host)
   end
 
   def csv_columns
-    [:name, :operatingsystem, :compute_resource_or_model, :hostgroup, :last_report]
+    User.current.table_preferences.find_by(name: controller_name)&.columns&.map { |col| col == 'os_title' ? 'operatingsystem' : col } - ['power_status']
```
Comment (adamruzicka) [inline app/controllers/hosts_controller.rb]: Why is power status hardcoded to be excluded?

Comment (pkoprda) [reply app/controllers/hosts_controller.rb]: I don't think that it's something necessary to be exported. And it always stays in the first column in exported file which tbh does not look good.

Comment (ares) [reply app/controllers/hosts_controller.rb]: OTOH the goal of the feature is consistency with what the user see on the page, if users hide the column in the view, they wouldn't get it even in the export, right?

Comment (pkoprda) [reply app/controllers/hosts_controller.rb]: Make sense, power column is now included in exported file.

File: app/controllers/hosts_controller.rb [hunk 2]
```diff
@@ -912,7 +912,7 @@ def host_attributes_for_templates(host)
   end
 
   def csv_columns
-    [:name, :operatingsystem, :compute_resource_or_model, :hostgroup, :last_report]
+    User.current.table_preferences.find_by(name: controller_name)&.columns&.map { |col| col == 'os_title' ? 'operatingsystem' : col }
```
Comment (adamruzicka) [inline app/controllers/hosts_controller.rb:915]: Is it possible that any other column might need to have its name mapped to something else? If so, could that be a part of the column definition?

File: test/controllers/hosts_controller_test.rb [hunk 3]
```diff
@@ -39,12 +39,13 @@ def host_attributes(host)
   end
 
   test "should get csv index with data" do
-    host = FactoryBot.create(:host, :with_hostgroup, :on_compute_resource, :with_reports)
+    User.current.table_preferences.create(name: 'hosts', columns: ['name', 'os_title', 'owner', 'hostgroup', 'last_report'])
+    host = FactoryBot.create(:host, :with_hostgroup, :with_reports)
     get :index, params: { :format => 'csv', :search => "name = #{host.name}" }, session: set_session_user
     assert_response :success
     buf = response.stream.instance_variable_get(:@buf)
-    assert_equal "Name,Operatingsystem,Compute Resource Or Model,Hostgroup,Last Report\n", buf.next
```
Comment (ofedoren) [inline test/controllers/hosts_controller_test.rb:46]: Just to not break previous CSV reports, I guess it's better to leave and test this entry as well.

**Review by ofedoren — CHANGES_REQUESTED** [review]
I've noticed that the all headers are exported, but the data for some of them are missing, e.g. content related (lifecycle env/content view) or power status (I'm not sure we previously exported it though) and some of the reported data column (but this can be an issue on my side)

File: app/controllers/hosts_controller.rb (hunk 2 above)
Comment (adamruzicka) [reply app/controllers/hosts_controller.rb:915]: Well, let's cross that bridge when we come to it.

**Comment by adamruzicka** [comment]
I'm taking over in https://github.com/theforeman/foreman/pull/9618 , thank you for your contribution @pkoprda !

**Pull request closed without merging**